query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -155,6 +317,13 @@ public void handleRequest(
       RequestAnalyzer requestAnalyzer) {
     long start = System.currentTimeMillis();
     validateQueryRequest(queryContext, request);
+
+    // Remove the filters which match the view column filters because they shouldn't be passed down
```

Comment:
That's not strictly true. Take the case of a view filtered to `FOO IN (X, Y)` and a filter provided that matches the view, `FOO = X` - it needs to be rewritten to `FOO = Y` (or `FOO IN (Y)`) when calling pinot. We should either handle that case, or remove the subset match logic and only take exact matches.

Reply:
Fixed this and added a test case to cover this case.